propagation.go

```diff
@@ -115,9 +126,9 @@ func (p *textMapPropagator) Inject(
 	// Do not encode the string with trace context to avoid accidental double-encoding
 	// if people are using opentracing < 0.10.0. Our colon-separated representation
 	// of the trace context is already safe for HTTP headers.
-	textMapWriter.Set(TracerStateHeaderName, sc.String())
+	textMapWriter.Set(p.headerKeys.TracerStateHeaderName, sc.String())
 	for k, v := range sc.baggage {
-		safeKey := addBaggageKeyPrefix(k)
+		safeKey := addBaggageKeyPrefix(k, p.headerKeys.TraceBaggageHeaderPrefix)
```

Comment:
instead of passing parameters, can we just change the functions to have the textMapWriter receiver?